### PR TITLE
Fix rollout when the collector config changes

### PR DESCRIPTION
**Pull request by bhiravabhatla**
Signed-off-by: santosh <bsantosh@thoughtworks.com>

Use pod annotations instead of deployment annotations to store SHA256 for config. This would help restart opentelemetry collector whenever config is updated.

Fixes https://github.com/open-telemetry/opentelemetry-operator/issues/460

### CODE REVIEW

**Comment by jpkrohling** [comment]
The e2e tests seem to be failing. 

@pavolloffay, once this is ready, can you review it?

**Comment by bhiravabhatla** [comment]
Fixed failing test @jpkrohling  @pavolloffay

File: pkg/collector/annotations.go [hunk 1]
```diff
@@ -37,13 +37,28 @@ func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
...
 	// make sure sha256 for configMap is always calculated
 	annotations["opentelemetry-operator-config/sha256"] = getConfigMapSHA(instance.Spec.Config)
 
 	return annotations
 }
 
 func getConfigMapSHA(config string) string {
-	h := sha256.Sum256([]byte(config))
-	return fmt.Sprintf("%x", h)
+	if config != "" {
```
Comment (pavolloffay) [inline pkg/collector/annotations.go:59]: any reason why there is an empty check? 

what retunrs the hash func on an empty string?

File: pkg/collector/annotations_test.go [hunk 2]
```diff
@@ -42,7 +42,6 @@ func TestDefaultAnnotations(t *testing.T) {
 	assert.Equal(t, "true", annotations["prometheus.io/scrape"])
 	assert.Equal(t, "8888", annotations["prometheus.io/port"])
 	assert.Equal(t, "/metrics", annotations["prometheus.io/path"])
-	assert.Equal(t, "9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08", annotations["opentelemetry-operator-config/sha256"])
```
Comment (pavolloffay) [inline pkg/collector/annotations_test.go:45]: why is this removed?

**Review by pavolloffay — COMMENTED** [review]
I would suggest adding an e2e test for this:
1. deploy collector as deployment
2. assert it is deployed
3. change config in CR - e.g. add new receiver
4. assert the collector pod restarted with the new changes - check if the new receiver port is exposed.

**Comment by VineethReddy02** [comment]
@bhiravabhatla can we move this PR forward? Also, do you mind replying back to the review comments?

Thanks!
